guide/interactions/buttons.md

```diff
@@ -0,0 +1,198 @@
+
+```js {2}
+client.on('interaction', interaction => {
+	if (!interaction.isButton()) return;
+	console.log(interaction);
+});
+```
+
+
+## Button collectors
+
+These work quite similarly to message collectors, except that you apply them on a message rather than a channel.
+
+### ButtonInteractionCollector
+
+To create a basic event-based `ButtonInteractionCollector` simply do as follows:
+
+```js
+client.on('message', message => {
+	const filter = interaction => interaction.customID === 'primary' && interaction.user.id === '122157285790187530';
+
+	const collector = message.createButtonInteractionCollector(filter, { time: 15000 });
+	collector.on('collect', i => console.log(`Collected ${i.customID}`));
+	collector.on('end', collected => console.log(`Collected ${collected.size} items`));
+});
+```
+
+### awaitButtonInteractions
+
+As with other types of collectors, you can also use a promise-based collector like this:
+
+```js {4-6}
+client.on('message', message => {
+	const filter = interaction => interaction.customID === 'primary' && interaction.user.id === '122157285790187530';
+
+	message.awaitButtonInteraction(filter, { time: 15000 })
+		.then(collected => console.log(`Collected ${collected.size} interactions`))
+		.catch(console.error);
+});
+```
+
+
+## Responding to buttons
+
+The `ButtonInteraction` class has similar methods as the `CommandInteraction` class, which we we'll be covering in the following section:
+
+::: tip
+The following methods behave exactly the same as on the `CommandInteraction` class:
+- `reply()`
+- `editReply()`
+- `defer()`
+- `fetchReply()`
+- `deleteReply()`
+- `followUp()`
+:::
+
+### Updating the button message
+
+The `ButtonInteraction` class provides a method to update the message the button is attached to, by using `ButtonInteraction#update()` as follows:
+
+```js {1,3}
+client.on('interaction', async interaction => {
+	if (!interaction.isButton()) return;
+	if (interaction.customID === 'primary') await interaction.update('A button was clicked', { components: [] });
+});
+```
+
+### Deferring and updating the button message
+
+Additionally to deferring the response of the interaction, you can defer the button, which will trigger a loading state:
+
+```js {1,5-9}
+const wait = require('util').promisify(setTimeout);
+
+client.on('interaction', async interaction => {
+	if (!interaction.isButton()) return;
+	if (interaction.customID === 'primary') {
+		await interaction.deferUpdate();
+		await wait(4000);
+		await interaction.update('A button was clicked', { components: [] });
+	}
+});
+```
+
+
+## Button styles
+
+Currently there are five different button styles available:
+
+* `PRIMARY` a blurple button
+* `SECONDARY` a grey button
+* `SUCCESS` a green button
+* `DANGER` a red button
+* `LINK` a button that navigates to a URL
+
```

Comment:
Assuming there will be a component mock-up around here once available.

Reply:
Yes, I will place a comment there for clarification